submodules/faxbox/views/edit.html

```diff
@@ -73,7 +73,20 @@ <h3>{{ i18n.callflows.faxbox.notification_emails }}</h3>
 				<div class="clearfix">
 					<label for="caller_id">{{ i18n.callflows.faxbox.caller_id }}</label>
 					<div class="input">
-						<input class="span4" id="caller_id" name="caller_id" type="text" placeholder="{{ i18n.callflows.faxbox.caller_id_placeholder }}" value="{{faxbox.caller_id}}" rel="popover" data-content="{{ i18n.callflows.faxbox.caller_id_data_content }}"/>
+						{{#if ensure_valid_caller_id}}
+							<span rel="popover" data-content="{{ i18n.callflows.faxbox.caller_id_data_content }}">
+								<select name="caller_id" id="caller_id" class="medium callflows-caller-id-dropdown">
+									{{#select faxbox.caller_id}}
+										<option value="_disabled">{{ i18n.callflows.faxbox.caller_id_no_selected }}</option>
```

Comment:
Now that you have an empty value to check against (`_disabled`), you should see where the faxbox data is normalized before being the request to update/save is made.

There, you can add your check to remove/update the `caller_id` prop from the faxbox object.